components/disk/component.go

```diff
@@ -244,13 +252,20 @@ func (c *component) Check() components.CheckResult {
 	}
 
 	for target := range c.mountPointsToTrackUsage {
-		if _, err := os.Stat(target); err != nil {
+		timeoutCtx, cancel := context.WithTimeout(c.ctx, 5*time.Second)
```

Comment:
@eahydra This is where we pass root context + timeouts, so we don't leak goroutines, in case of blocking stat operations.